### DISCUSSION TITLE
TpmPrivate buffer

### DISCUSSION THREAD

**Discussion by GmanPilot** [microsoft/TSS.MSR#135] [Q&A]
When I create a ECDSA key, 
the "**tpm[parentKeyAuth].Create** " method returns a TpmPrivate object.

Two questions:
1) Am I correct in assuming this is the private key portion of the newly created key?
2) If so, what is the format?

I examined the buffer of the TPMPrivate object and it is 94 bytes in length, however I do not see the "X" and "Y" values in there.
I exported the public key after creation and compared them to it and nowhere in the private object are the "X" and "Y" points found. This leads me to belive it only contains the "D" point. However The buffer does not appear to be DER encoded because I could not find what the 00 20 identifer referred to.

Thanks for your help.

Here is some sample data I created for a reference:

`PrivateKey buffer: 00 20 37 f7 70 9e 14 21 ef 7b 88 c2 3c 3e f9 ca 59 aa d7 72 3c 0a 33 c8 c7 d3 c5 17 b6 8a 8e 7a 39 a7 00 10 62 65 c7 df a7 e2 54 db 7b ba 60 c8 ac a7 70 7f 8f 07 8b 97 3f ef e3 88 7e cc 96 cd 69 dd a5 37 71 8a a1 f6 26 c9 9e 5f 7d d6 13 d4 ad 78 58 04 2c 8d 59 6c 9c f7 d6 c5 b6 ac

X = 55 1d 67 7c c0 7b 1b 54 61 18 cc ea 41 2e 05 46 f0 61 f2 20 03 07 e3 c3 54 a6 13 23 31 fd 72 71
Y = 1d cb d1 e8 7b c7 7c d0 18 44 57 c1 76 4c 1e 64 8d 6c df f4 41 2a 3c 93 47 17 65 74 5e 29 9d ec
`

**Comment by RonaldAi** [comment]
The format of the TPM2B_PRIVATE is explained in TPM spec, Part 2, Section 12.3.7 (https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf) 
That also explains that a type that starts with "TPM2B" has two bytes (2B) at the beginning that contain the size of the data.
